ffi/src/post_impl.rs

```diff
@@ -73,24 +75,68 @@ fn _generate_proof(
     let challenge = unsafe { std::slice::from_raw_parts(challenge, 32) };
     let challenge = challenge.try_into()?;
 
-    let proof = prove::generate_proof(datadir, challenge, cfg, nonces, threads)?;
+    let proof = prove::generate_proof(datadir, challenge, cfg, nonces, threads, pow_flags)?;
 
     let mut indices = ManuallyDrop::new(proof.indices);
     let (ptr, len, cap) = (indices.as_mut_ptr(), indices.len(), indices.capacity());
     let proof = Box::new(Proof {
         nonce: proof.nonce,
         indices: ArrayU8 { ptr, len, cap },
-        k2_pow: proof.k2_pow,
+        pow: proof.pow,
     });
 
     Ok(proof)
 }
 
 #[repr(C)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum VerifyResult {
     Ok,
     Invalid,
     InvalidArgument,
+    FailedToCreateVerifier,
+}
+
+/// Get the recommended RandomX flags
+///
+/// Does not include:
+/// * FLAG_LARGE_PAGES
+/// * FLAG_FULL_MEM
+/// * FLAG_SECURE
+///
+/// The above flags need to be set manually, if required.
+#[no_mangle]
+pub extern "C" fn recommended_pow_flags() -> post::pow::randomx::RandomXFlag {
+    post::pow::randomx::RandomXFlag::get_recommended_flags()
+}
+
+#[no_mangle]
+pub extern "C" fn new_verifier(
+    flags: post::pow::randomx::RandomXFlag,
+    out: *mut *mut Verifier,
```

Comment:
why do you prefer this approach over returning tuple with Verifier and result?

Reply:
It's not possible to return a tuple:
>cbindgen sadly cannot ever support anonymous tuples (A, B, ...), as there is no way to guarantee their layout. You must use a tuple struct.

 I would need to return a struct like:
```rust
#[repr(C)]
struct NewVerifierResult {
    verifier: *mut Verifier,
    result: VerifyResult,
}

#[no_mangle]
pub extern "C" fn new_verifier(
    flags: post::pow::randomx::RandomXFlag,
) -> NewVerifierResult {
```

It's a convention in C to return created things via a `Type **out` and the error code in the return value.

Reply:
yes, i meant struct with two fields, not actual tuple